packages/salesforcedx-vscode-core/src/orgBrowser/orgMetadata.ts

```diff
@@ -0,0 +1,139 @@
+    const startTime = process.hrtime();
+    const cancellationTokenSource = new vscode.CancellationTokenSource();
+    const cancellationToken = cancellationTokenSource.token;
+
+    const execution = new CliCommandExecutor(this.build(response.data), {
+      cwd: getRootWorkspacePath()
+    }).execute(cancellationToken);
+
+    execution.processExitSubject.subscribe(async data => {
+      this.logMetric(execution.command.logName, startTime);
+      await buildTypeList();
+    });
+    notificationService.reportExecutionError(
+      execution.command.toString(),
+      (execution.stderrSubject as any) as Observable<Error | undefined>
+    );
+    channelService.streamCommandOutput(execution);
+    ProgressNotification.show(execution, cancellationTokenSource);
+    taskViewService.addCommandExecution(execution, cancellationTokenSource);
+  }
+}
+
+async function getDirectory(): Promise<string | undefined> {
+  if (hasRootWorkspace()) {
+    const workspaceRootPath = getRootWorkspacePath();
+    const defaultUsernameOrAlias = await getDefaultUsernameOrAlias();
+    const defaultUsernameIsSet = typeof defaultUsernameOrAlias !== 'undefined';
+
+    if (defaultUsernameIsSet) {
+      const username = await OrgAuthInfo.getUsername(defaultUsernameOrAlias!);
+      const metadataTypesPath = path.join(workspaceRootPath, '.sfdx', 'orgs', username, 'metadata', 'metadataTypes.json');
+      return metadataTypesPath;
+    } else {
+      throw new Error(nls.localize('error_no_default_username'));
+    }
+  } else {
+    throw new Error(nls.localize('cannot_determine_workspace'));
+  }
+}
+
+export async function getDefaultUsernameOrAlias(): Promise<string | undefined> {
+  if (hasRootWorkspace()) {
+    return await OrgAuthInfo.getDefaultUsernameOrAlias();
+  }
+}
+
+const workspaceChecker = new SfdxWorkspaceChecker();
+const parameterGatherer = new EmptyParametersGatherer();
+
+export async function forceDescribeMetadata() {
+  const outputPath = await getDirectory();
+  if (!isNullOrUndefined(outputPath)) {
+    const describeExecutor = new ForceDescribeMetadataExecutor(outputPath);
+    const commandlet = new SfdxCommandlet(
+      workspaceChecker,
+      parameterGatherer,
+      describeExecutor
+    );
+    await commandlet.run();
+  }
+
+}
+
+export type MetadataObject = {
+  directoryName: string;
+  inFolder: boolean;
+  metaFile: boolean;
+  suffix: string;
+  xmlName: string;
+};
+
+export async function buildTypeList() {
+  const filePath = await getDirectory();
+  if (!isNullOrUndefined(filePath)) {
+    const fileData = JSON.parse(fs.readFileSync(filePath, 'utf8'));
+    const metadataObjects = fileData.metadataObjects;
+    const metadataTypes = [];
+    for (const index in metadataObjects) {
+      if (!isNullOrUndefined(metadataObjects[index].xmlName)) {
+        metadataTypes.push(metadataObjects[index].xmlName);
+      }
+    }
+  } else {
+    throw new Error('There was an error retrieving metadata type information.');
```

Comment:
@ruthemmanuelle This error would get thrown if there is an issue finding the file with the metadata type information, do you think this error message should be more explicit?

Reply:
Whenever possible, we try to make error messages include suggested next steps. What should the user do to recover?

Reply:
Also, like the other message, I believe this should be in a messages file.

Reply:
@AnanyaJha - I saw your question on this PR, "Assuming that retrieving the metadata types runs in the background whenever the default org is switched - there are a number of places where errors can be thrown ie. no workspace, an issue with finding the default username, error reading the file. Do we want to actively throw those errors or just log them with a console.error(**)?" When these errors occur, is there anything the user should do to avoid problems? Or will we have another opportunity to communicate with the user before they actually try to work with the retrieved types about whatever issues they need to address?